analyses/filter-mutation-frequencies-tables/filter-mutation-frequencies-tables.Rmd

```diff
@@ -0,0 +1,309 @@
+# read file
+gtf <- file.path(data_dir, "gencode.v38.primary_assembly.annotation.gtf.gz")
+gencode_gtf <- rtracklayer::import(con = gtf) %>% as.data.frame() %>% tibble()
+```
+
+Get Ensembl identifiers
+```{r}
+# filter for gene rows and get ids
+ensg_ids <- gencode_gtf %>% 
+  dplyr::filter(type == "gene") %>% 
+  dplyr::mutate(gene_id = stringr::str_extract(gene_id, "ENSG\\d+")) %>% 
+  pull(gene_id) %>% 
+  unique() 
+ensg_ids <- ensg_ids[!is.na(ensg_ids)]
+
+# Remove gencode dataframe
+rm(gene_code_annot)
+```
+
+Get OpenPedCan sample IDs (Kids_First_Biospecimen_ID)
+```{r}
+sample_ids <- readr::read_tsv(file.path(data_dir, "histologies.tsv"), 
+                              guess_max = 10000, show_col_types = FALSE) %>% 
+  dplyr::filter(sample_type == "Tumor") %>% 
+  dplyr::select(Kids_First_Biospecimen_ID, cohort)
+```
+
+## Filter mutation frequencies tables
+
+Function to filter for required Ensembl gene identifiers and write to file
+```{r}
+
+filter_freq_table <- function(freq_file, ensg_ids) {
+  
+  # Read gene level table 
+  freq_table <-  readr::read_tsv(freq_file, guess_max = 10000, 
+                                 show_col_types = FALSE) 
+  
+  # Write filtered TSV file
+  file_name <- basename(freq_file)
+  freq_table %>% dplyr::filter(targetFromSourceId %in% ensg_ids) %>% 
+  readr::write_tsv(file.path(results_dir, file_name))
+
+  # Write filtered JSON file
+  json_file <- paste(unlist(str_split(file_name, "\\."))[1], "json", sep = ".")
+  freq_table %>% dplyr::filter(targetFromSourceId %in% ensg_ids) %>% 
+    jsonlite::write_json(file.path(results_dir,  json_file))  
+
+return(freq_table)
+}
+```
+
+#### Gene level SNV frequencies
+```{r}
+# Remove Ensembl gene identifiers
+gene_level_snv <- filter_freq_table(
+  file.path(snv_results_dir, 
+            "gene-level-snv-consensus-annotated-mut-freq.tsv.gz"), ensg_ids) 
+
+# Ensembl gene identifiers removed
+filtered_ensg <- gene_level_snv %>% 
+  dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
+  dplyr::select(Gene_symbol, targetFromSourceId) %>%
+  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
+  dplyr::distinct()
+
+# read the SNV consensus file
+consensus_ensg <- data.table::fread(
+  file.path(data_dir, "snv-consensus-plus-hotspots.maf.tsv.gz"), 
+  select = c("Tumor_Sample_Barcode", "Gene"), showProgress = FALSE) %>% 
+  tibble() %>% 
+  dplyr::filter(!is.na(Gene)) %>% 
+  dplyr::rename(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode) %>% 
+  dplyr::distinct()
+
+# merge sample IDs to the ENSG IDs not in gencode v38
+filtered_ensg <- filtered_ensg %>% 
+  dplyr::left_join(consensus_ensg, by=c("Ensembl_ID" = "Gene"))
+
+# merge OpenPedCan cohorts to the ENSG IDs not in gencode v38
+filtered_ensg <- filtered_ensg %>% 
+  dplyr::left_join(sample_ids, by= "Kids_First_Biospecimen_ID")
+
+# write to file
+readr::write_tsv(filtered_ensg,
+  file.path(results_dir, 
+            "gene-level-snv-consensus-annotated-mut-freq_dropped_ensg.tsv.gz"))
+filtered_ensg
```

Comment:
For this output - could you please do:
```
fitlered_ensg %>% select(Gene_symbol, Ensembl_ID, cohort) %>% table()
```
Just to see what are the genes that are affected and we can keep all the individual samples output to tsv files as you have done here.

Reply:
> For this output - could you please do:
> 
> ```
> fitlered_ensg %>% select(Gene_symbol, Ensembl_ID, cohort) %>% table()
> ```
> 
> Just to see what are the genes that are affected and we can keep all the individual samples output to tsv files as you have done here.

@runjin326, sorry that am understanding what you are suggesting. All the tables are printed in an RMD file. Please open this attached.  Is there any need to use `fitlered_ensg %>% select(Gene_symbol, Ensembl_ID, cohort) %>% table()` instead of simply `fitlered_ensg`. These three columns are the only columns in the tables. Any advantage to select them rather than just display the whole tables as rendered in this attached HTML file. 
[filter-mutation-frequencies-tables.nb.html.zip](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/files/7889758/filter-mutation-frequencies-tables.nb.html.zip)

Reply:
> 

@runjin326,  I get your point now.  You want to exclude `Kids_First_Biospecimen_IDs` and only print them out to file because make the tables much larger.  Sorry for the misunderstanding! I will ASAP